flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -177,11 +178,13 @@ private JobGraph createJobGraph() {
 
 		setSlotSharingAndCoLocation();
 
+		// For now, only consider managed memory for batch algorithms.
+		// TODO: extend managed memory fraction calculations w.r.t. various managed memory use cases.
 		setManagedMemoryFraction(
 			Collections.unmodifiableMap(jobVertices),
 			Collections.unmodifiableMap(vertexConfigs),
 			Collections.unmodifiableMap(chainedConfigs),
-			id -> streamGraph.getStreamNode(id).getManagedMemoryWeight());
+			id -> streamGraph.getStreamNode(id).getManagedMemoryUseCaseWeights().getOrDefault(ManagedMemoryUseCase.BATCH_OP, 0));
```

Comment:
Not sure about this. I think we don't really want a default weight here.
The purpose for this default is to avoid NPE when performing `mapToInt` and `sum` in `setManagedMemoryFractionForSlotSharingGroup`.
Maybe we should not set default value here and handles the `null` values in `setManagedMemoryFractionForSlotSharingGroup`. I think it should be straightforward to convert `null` to `0` for a `sum` processing.